pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go

```diff
@@ -93,8 +137,26 @@ func (fam *RemoteSecondaryForkedAlertmanager) CleanUp() {
 }
 
 func (fam *RemoteSecondaryForkedAlertmanager) StopAndWait() {
+	// Stop the internal Alertmanager.
 	fam.internal.StopAndWait()
+	// Stop our alert senders.
 	fam.remote.StopAndWait()
+
+	// Send config and state to the remote Alertmanager.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	if err := fam.remote.CompareAndSendState(ctx); err != nil {
+		fam.log.Error("Error sending state to the remote Alertmanager", "err", err)
+	}
+
+	config, err := fam.store.GetLatestAlertmanagerConfiguration(ctx, fam.orgID)
+	if err != nil {
+		fam.log.Error("Error getting latest Alertmanager configuration", "err", err)
+		return
+	}
+	if err := fam.remote.CompareAndSendConfiguration(ctx, config); err != nil {
+		fam.log.Error("Error sending configuration to the remote Alertmanager", "err", err)
+	}
```

Comment:
I don't think we need this to be a comparison - shouldn't we just send whatever we have at this point? What's your reasoning for the comparison?

Reply:
1. If we have the same on both Alertmanagers, there's no need to create a new configuration in the remote Alertmanager. The same goes for the state.
2. We'd need to add public methods for sending the configuration without checking first.

Ultimately I think this should follow the same logic as `ApplyConfig()`, I don't see any difference between what we want to achieve here and on every call to `ApplyConfig()`. If we were to remove the comparison here, we should do the same in the other method.

Reply:
I created a [PR to remove this comparison](https://github.com/grafana/grafana/pull/79469), I don't think it makes much sense to keep it around. We can use that change in this PR once it's merged.